File: Bit Manipulation/EvenOrOdd/evenorodd.go
package main

import (
	"fmt"
)

func main() {
	for i := 0; i < 10; i++ {
		if i&1 == 1 {
			fmt.Println(i, "is odd")
		} else {
			fmt.Println(i, "is even")
		}
	}
}

File: Bit Manipulation/EvenOrOdd/even_or_odd.c
#include <stdio.h>

void main(void){
	
	//quick test to see if number is even or odd
	//lsb toggles every time you increment a number
	//0 - 0000 - even
	//1 - 0001 - odd
	//2 - 0010 - even
	//3 - 0011 - odd
	//4 - 0100 - even 
	//5 - 0101 - odd
	//etc
	//pretty dope
	for(int evenodd = 0; evenodd <= 10; evenodd++){
		if(evenodd & 0x1){
			printf("%i is odd\n", evenodd);
		}else{
			printf("%i is even\n", evenodd);
		}
	}
	
}
